### ISSUE TITLE
publishDir pattern '{"*.bam", "*len.txt"}'  is not publishing the output files

### ISSUE THREAD

**Issue by kemin711** [nextflow-io/nextflow#5327]
Maybe I should use slack.  Next time I will try slack.  I might have simple typo errors that prevented the publishing. Or this could be a migration issue.  It seemed work with DSL1. Now I am using DSL2, my files are not published.

**Comment by bentsherman** [comment]
Can you share a small example that I can test?

**Comment by bentsherman** [comment]
Here are some other ways to write that pattern that might work:
```groovy
'{*.bam,*len.txt}'
'*{.bam,len.txt}'
```

**Comment by kemin711** [comment]
I did some experiment using my workflow.  1. when using '{"*.bam", "*xyz.txt"}' format, then nothing will be published. 2. without using the double for each element, then only the first one will be published.  
I looked at the glob document in java, looks you don't need the double quote for each element.  It could a a bug in nextflow.  
My current workaround is to use 

`publishDir: "targetdir", mode: 'rellink', saveAs: { fname -> { if (fname.endsWith("blabla") || fname.endsWith(".bam")) { fname } else { null } }`
